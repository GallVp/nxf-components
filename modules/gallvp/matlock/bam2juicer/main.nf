process MATLOCK_BAM2JUICER {
    tag "$meta.id"
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/matlock:20181227--h665f8ca_8':
        'biocontainers/matlock:20181227--h665f8ca_8' }"

    input:
    tuple val(meta), path(bam)

    output:
    tuple val(meta), path("*.links.txt")    , emit: links_txt
    path "versions.yml"                     , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def prefix = task.ext.prefix ?: "${meta.id}"
    def VERSION = '20181227'
    """
    matlock \\
        bam2 \\
        juicer \\
        $bam \\
        ${prefix}.links.txt

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        matlock: $VERSION
    END_VERSIONS
    """

    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"
    def VERSION = '20181227'
    """
    touch ${prefix}.links.txt

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        matlock: $VERSION
    END_VERSIONS
    """
}
