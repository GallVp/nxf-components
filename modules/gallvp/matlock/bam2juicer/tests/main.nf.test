nextflow_process {

    name "Test Process MATLOCK_BAM2JUICER"
    script "../main.nf"
    process "MATLOCK_BAM2JUICER"

    tag "modules"
    tag "modules_gallvp"
    tag "matlock"
    tag "matlock/bam2juicer"

    test("sarscov2 - bam - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/bam/test.paired_end.sorted.bam', checkIfExists: true),
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}