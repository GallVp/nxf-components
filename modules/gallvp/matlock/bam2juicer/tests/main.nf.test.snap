{
    "sarscov2 - bam - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.links.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,3a7db756612e49e49d57eb3f3160b955"
                ],
                "links_txt": [
                    [
                        {
                            "id": "test"
                        },
                        "test.links.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,3a7db756612e49e49d57eb3f3160b955"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-03-12T12:40:21.02834"
    }
}